runtime-sdk/modules/evm/src/lib.rs

```diff
@@ -625,6 +616,37 @@ impl<Cfg: Config> Module<Cfg> {
         derive_caller::from_tx_auth_info(ctx.tx_auth_info())
     }
 
+    /// Returns the decrypted call data or `None` if this transaction is simulated in
+    /// a context that may not include a key manager (i.e. SimulateCall but not EstimateGas).
+    fn decode_call_data<C: Context>(
+        ctx: &C,
+        data: Vec<u8>,
+        format: transaction::CallFormat,
+        tx_index: usize,
+        assume_km_reachable: bool,
+    ) -> Result<Option<(Vec<u8>, callformat::Metadata)>, Error> {
+        if !Cfg::CONFIDENTIAL || format != transaction::CallFormat::Plain || data.is_empty() {
```

Comment:
I moved this around. The only change is `|| data.is_empty()`

Reply:
do we also need encryption for 'create' calls?

Reply:
did I miss anything here? Shouldn't it be `format == transaction::CallFormat::Plain`, then we return the data immediately?

Reply:
sapphire treats plain/unencrypted as encrypted, for compatibility with an unmodified eth envelope